### DISCUSSION TITLE
Copying files into a container vs using a volume

### DISCUSSION THREAD

**Discussion by sergiorivas** [BretFisher/ama#141] [Q&A]
When I am creating a `Dockerfile` I wonder if is really necessary to do

```
COPY . $APPDIR 
```
 If in the future I will use a `-v $(pwd):/app_dir`

Does this make any sense for (developing mode)?

**Comment by BretFisher** [comment]
For pure local development where you're always bind-mounting, no the `COPY` isn't needed. However:

1. You ideally want to use the same Dockerfile for server images as well as development, so you'll end up having the `COPY` line in the Dockerfile anyway.
2. If you're bind-mounting the correct stuff, then you rarely need to rebuild the image locally anyway, so it's not going to save much time/space usually.
